src/packet.rs

```diff
@@ -202,8 +202,8 @@ impl Packet {
         let report = poc_lora::LoraWitnessReportReqV1 {
             pub_key: vec![],
             data: payload,
+            tmst: self.timestamp as u32,
             timestamp: self.timestamp,
```

Comment:
self timestamp is 2 fields here? `tmst` should be the `tmst` from the rxpk packet and `timestamp` should be the system timestamp in nanoseconds?

Reply:
yea I think the `Packet` needs to have the fields added and then the `impl From<RxPk> for Packet` will need to use rxpk.get_timestamp() to get tmst and rxpk.get_time() to get UTC string. Apologies again for the crappy function names.

Reply:
Can you confirm we’re talking about a `push_data::RxPk` here? In which case the timestamp field we’re creating _is_ the tmst if I read this line correctly: https://github.com/helium/gateway-rs/blob/0fae27b7939e2051e02b56eaffef09ac4dd3582d/src/packet.rs#L53

in which case I need to add the tmst field to the proto and then re-assign this `*rxpk.get_timestamp` to that field and use the other time getter for the existing field, right?

Reply:
Yes, your understanding is correct, however, you'll want to check normal uplinks bc they might have logic hanging off of Packet.timestamp that assumes it's the 32-bit counter (which impacts downlink scheduling).

Reply:
One idea is to leave `Packet.timestamp` as is and to create `Packet.datetime` which will take the UTC string which is the new data.

Reply:
That’s a good suggestion; will do

Reply:
comment added to get the proper datetime value added to packet.proto and then get the necessary value from rxpk and convert accordingly as a future todo; temporary solution to set the field only on the witness report from the receiving gateway's systemtime::now in place